src/firmware/app/control/trajectory_planner.h

```diff
@@ -72,232 +71,369 @@ typedef enum TrajectoryPlannerGenerationStatus
+ * the total displacement of each segment
+ *
+ * @pre All arrays must be pre-allocated up to at least
+ * TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS
+ *
+ * @param segment_lengths [in] The length of each segment in meters. The segment length is
+ * the distance travelled between successive speed elements.
+ *
+ * @param speeds [in] The speed profile defining the current speed at each element. In
+ * m/s.
+ *
+ * @param trajectory_duration [in] The duration between successive speed elements in the
+ * profile. In seconds.
+ *
+ * @param num_elements [in] The number of elements in the speed profile.
+ */
+void app_trajectory_planner_generatePositionTrajectoryTimeProfile_2(
+    float segment_lengths[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS],
+    float speeds[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS],
+    float trajectory_duration[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS],
+    unsigned int num_elements);
 
 /**
- * Modifies a forwards continuous velocity profile to also be backwards continuous based
- * on the input parameters.
+ * Function balances the segment time durations of 2 separate speed profiles. The segment
+ * durations of profile1 and profile2 will be made equal - to the LONGEST duration at each
+ * point along the profiles. This is done by modifying the final speed elements.
  *
- * NOTE: This function does not break forwards continuity of the trajectory
+ * @pre The profile must have the same number of elements. Other than the displacement
+ * arrays that must have 1 fewer element.
  *
- * @pre The input velocity_profile is required to be FORWARDS CONTINUOUS before it is used
- * as an input to this function
- * @pre forwards_continuous_velocity_profile is pre-allocated up to at least
- * TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS
+ * @pre The arrays must be pre-allocated up to at least
+ * TRAJECTORY_PLANNER_MAX_NUM_SEGMENTS.
+ *
+ * @param speeds1 [in] The speed profile corresponding to trajectory 1. In m/s.
  *
- * @param num_segments [in] The number of segments(elements) in the velocity_profile array
- * @param forwards_continuous_velocity_profile [in/out] The forwards continuous velocity
- * profile to ebe modified in-place into a profile that is also backwards continuous
- * @param arc_segment_length [in] The arc segment length of each segment in the path the
- * profile is being generated for in meters
- * @param max_allowable_acceleration [in] The max allowable acceleration at any point on
- * the velocity profile
+ * @param speeds2 [in] The speed profile corresponding to trajectory 2. In m/s.
+ *
+ * @param durations1 [in] The time durations corresponding to the length of time between
+ * successive speed points in trajectory 1. In seconds.
+ *
+ * @param durations2 [in] The time durations corresponding to the length of time between
+ * successive speed points in trajectory 2. In seconds.
+ *
+ * @param displacement1 [in] The total displacement corresponding to the distance between
```

Comment:
`total displacement` confused be for a second here, because I thought `total` meant the displacement over the entire profile. I think just `displacement` is fine here because it's explained further later (that it's between successive elements)

Reply:
Removed 'total' 👍